tcp_test.go

```diff
@@ -35,7 +35,7 @@ func TestTCPConnection(t *testing.T) {
 	go func() {
 		conn, err := ln.Accept()
 		if err != nil {
-			t.Fatalf("Error accepting on socket: %s", err)
+			panic(fmt.Sprintf("Error accepting on socket: %s", err))
```

Comment:
Why are these changing? They look correct to me.

Reply:
```
the goroutine calls T.Fatalf, which must be called in the same goroutine as the test (SA2002)
```